midi_rag.py

```
@@ -0,0 +1,333 @@
+            # 음계 분석 (12개 음에 대한 히스토그램)
+            *[notes.count(i % 12) / len(notes) for i in range(12)]
+        ]
+        
+        return np.array(features, dtype=np.float32)
+    
+    def build_database(self):
+        """MIDI 폴더의 모든 파일을 분석하여 데이터베이스 구축"""
+        debug_log(f"MIDI 데이터베이스 구축 시작: {self.midi_folder}")
+        
+        midi_files = glob.glob(os.path.join(self.midi_folder, "*.mid")) + \
+                    glob.glob(os.path.join(self.midi_folder, "*.midi"))
```

Flake8 (7.2.0)

[error] continuation line over-indented for visual indent

(E127)